smart-contracts/test/testCosmosBridge.js

```diff
@@ -306,5 +284,124 @@ describe("Test Cosmos Bridge", function () {
         "10000.0000000000000001"
       );
     });
+
+    it("should deploy a new token upon the successful processing of a double-pegged burn prophecy claim", async function () {
+      state.nonce = 1;
+
+      const { digest, claimData, signatures } = await getValidClaim({
+        state,
+        sender: state.sender,
+        senderSequence: state.senderSequence,
+        recipientAddress: state.recipient,
+        tokenAddress: state.token.address,
+        amount: state.amount,
+        isDoublePeg: true,
+        nonce: state.nonce,
+        networkDescriptor: state.networkDescriptor,
+        tokenName: state.name,
+        tokenSymbol: state.symbol,
+        tokenDecimals: state.decimals,
+        validators: [userOne, userTwo, userFour],
+      });
+
+      const expectedAddress = ethers.utils.getContractAddress({ from: state.bridgeBank.address, nonce: 1 });
+
+      await expect(state.cosmosBridge
+        .connect(userOne)
+        .submitProphecyClaimAggregatedSigs(
+            digest,
+            claimData,
+            signatures
+        )).to.emit(state.cosmosBridge, 'LogNewBridgeTokenCreated')
+        .withArgs(18, state.networkDescriptor, state.name, state.symbol, state.token.address, expectedAddress);
+
+      const newlyCreatedTokenAddress = await state.cosmosBridge.sourceAddressToDestinationAddress(state.token.address);
+      expect(newlyCreatedTokenAddress).to.be.equal(expectedAddress);
+    });
+
+    it("should NOT deploy a new token upon the successful processing of a normal burn prophecy claim", async function () {
+      state.nonce = 1;
+
+      const { digest, claimData, signatures } = await getValidClaim({
+        state,
+        sender: state.sender,
+        senderSequence: state.senderSequence,
+        recipientAddress: state.recipient,
+        tokenAddress: state.token.address,
+        amount: state.amount,
+        isDoublePeg: false,
+        nonce: state.nonce,
+        networkDescriptor: state.networkDescriptor,
+        tokenName: state.name,
+        tokenSymbol: state.symbol,
+        tokenDecimals: state.decimals,
+        validators: [userOne, userTwo, userFour],
+      });
+
+      await state.cosmosBridge
+        .connect(userOne)
+        .submitProphecyClaimAggregatedSigs(digest, claimData, signatures);
+
+      const newlyCreatedTokenAddress = await state.cosmosBridge.sourceAddressToDestinationAddress(state.token.address);
+      expect(newlyCreatedTokenAddress).to.be.equal(state.constants.zeroAddress);
```

Comment:
We should also assert that the recipient's balance of the token went up by the amount we specified in the claim.